tests/unit/test_executor.py

```diff
@@ -1,12 +1,24 @@
-import logging
+from unittest.mock import MagicMock, patch
 
 from latigo.executor import PredictionExecutor
+from latigo.gordo import GordoModelInfoProvider
+from tests.factories.task import TaskFactory
 
-logger = logging.getLogger(__name__)
 
+@patch("latigo.gordo.prediction_execution_provider.GordoClientPool", new=MagicMock())
+@patch(
+    "latigo.executor.model_info_provider_factory", new=MagicMock(side_effect=MagicMock(spec_set=GordoModelInfoProvider))
+)
+@patch("latigo.executor.PredictionExecutor._perform_auth_checks", new=MagicMock())
```

Comment:
```suggestion
@patch("latigo.gordo.prediction_execution_provider.GordoClientPool")
@patch(
    "latigo.executor.model_info_provider_factory", side_effect=MagicMock(spec_set=GordoModelInfoProvider))
)
@patch("latigo.executor.PredictionExecutor._perform_auth_checks")
```
Default `new` argument is `MagicMock`. Also not sure what is the purpose of additional arguments to the patch of `"latigo.executor.model_info_provider_factory"`. Perhaps plain `@patch("latigo.executor.model_info_provider_factory", spec=GordoModelInfoProvider)` is enough.

Reply:
Also patching a private method looks wrong

Reply:
if you won't specify `new` then you should pass it as the argument and we don't need it in the function.
Plain won't work cause we're mocking a factory here.

Can't see nothing wrong in the private method mocking, sometimes like here you have to disable some inner logic and this is the way to do it. Why is it wrong from your perspective?